mvrx/src/main/kotlin/com/airbnb/mvrx/Async.kt

```diff
@@ -36,17 +36,13 @@ sealed class Async<out T>(val complete: Boolean, val shouldLoad: Boolean) {
     }
 }
 
-object Uninitialized : Async<Nothing>(complete = false, shouldLoad = true), Incomplete
+object Uninitialized : Async<Nothing>(complete = false, shouldLoad = true, value = null), Incomplete
 
-class Loading<out T> : Async<T>(complete = false, shouldLoad = false), Incomplete {
-    override fun equals(other: Any?) = other is Loading<*>
+data class Loading<out T>(private val value: T? = null) : Async<T>(complete = false, shouldLoad = false, value = value), Incomplete
 
-    override fun hashCode() = "Loading".hashCode()
-}
-
-data class Success<out T>(private val value: T) : Async<T>(complete = true, shouldLoad = false) {
+data class Success<out T>(private val value: T) : Async<T>(complete = true, shouldLoad = false, value = value) {
 
-    override operator fun invoke(): T = value
+    override fun invoke(): T = super.invoke() as T
```

Comment:
You can leave this as is. Returning value is still correct here

Reply:
Sure, it was a result of my testing of making `Success` not a data class which did not worked.